transformations/transformations/single_column_coalesced_vector.py

```diff
@@ -73,8 +73,11 @@ def extract_vector_sections(cls, section, horizontal):
         _scope_node_types = (ir.Loop, ir.Conditional, ir.MultiConditional)
 
         # Identify outer "scopes" (loops/conditionals) constrained by recursive routine calls
-        separator_nodes = []
         calls = FindNodes(ir.CallStatement).visit(section)
+        pragmas = [pragma for pragma in FindNodes(ir.Pragma).visit(section) if pragma.keyword.lower() == "loki" and
```

Comment:
If I have understood the use-case for `!$loki separator` pragmas correctly, I think the same functionality can be achieved by triggering the `trim_vector_sections` option on the driver loops too.

The only difficulty there is calling the dataflow analysis on `driver_loop` since loops are frozen ir Nodes and the dataflow analysis attacher wants to assign to `driver_loop.body`.

I did a quick hack and simply ran the dataflow analysis attacher as an in-place transform that doesn't return anything. That seems to work; the relevant `_live/_uses/_defines_symbols` sets were updates correctly.

Long story short: if you rely on this leave this in, but I'll probably look into replacing the `!$loki separator` functionality with `trim_vector_sections` once I've had the chance to understand the `DataflowAnalysisAttacher` a bit better.